src/basic/slow-code.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#pragma once
+
+#include "log.h"
+#include "string-util.h"
+#include "time-util.h"
+
+#include "macro.h"
+
+typedef struct slow_code_detector {
+        uint32_t timeout_sec;
+        uint32_t count;
+        usec_t current_time;
```

Comment:
instead of storing both current_time and timeout_sec, just store a single absolute timeout, i.e. `timeout_usec` which is initialized to `usec_add(now(CLOCK_MONOTONIC), relative_timeout_usec`